src/components/ImageCard.js

```diff
@@ -1,22 +1,40 @@
 import "bootstrap/dist/css/bootstrap.min.css";
 import React, { Component } from "react";
 import { Button, Card } from "react-bootstrap";
-
 export default class ImageCard extends Component {
   constructor(props) {
     super(props);
     this.state = {
       favorites: 0,
+      shouldShowModal: false,
+      modalImageSrc: "",
+      imageDescription: "",
+      title: "",
     };
   }
+  openModal = (title) => (props) => {
+    this.setState({ shouldShowModal: true });
+    this.setState({ modalImageSrc: props.target.src });
+    this.setState({ imageDescription: props.target.alt });
+
+    this.setState({ title: title });
+    console.log(title);
+  };
+
+  closeModal = () => {
+    this.setState({ shouldShowModal: false });
+    this.setState({ modalImageSrc: "" });
+    this.setState({ imageDescription: "" });
+    this.setState({ imageTitle: "" });
+  };
```

Comment:
- Tip here, not sure if I shared this already is that you can just use one object in `this.setState()` and pass all key values pairs that you need inside 👍  This way you don't have to write `this.setState()' over and over (saves some time and helps with readability).